src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -65,7 +65,6 @@ internal partial class EETypeNode : ObjectNode, ISymbolNode, IEETypeNode
 
         public EETypeNode(NodeFactory factory, TypeDesc type)
         {
-            Debug.Assert(!type.IsCanonicalSubtype(CanonicalFormKind.Specific));
```

Comment:
Please add a check that if the type is CanonicalSubType, that it is in correct canonical form like List<Canon>, and not something like List<List<Canon>>